src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -821,6 +821,22 @@ internal override bool HasSpecialName
             }
         }
 
+        internal override bool HasInternalImplementationOnlyAttribute
+        {
+            get
+            {
+                foreach (var att in this.GetAttributes())
```

Comment:
Please add a comment about why we think it is better to repeatedly iterate over attributes over an over again vs. caching the fact of the attribute presence similar to the other well-known attributes. Why it is a right thing to make an exception for this attribute.